src/hooks/useCart/useCart.tsx

```diff
@@ -296,28 +315,36 @@ export const useCart = (
     const checkout = async (): Promise<void> => {
       setCartState("CHECKING_OUT");
 
-      const allIdentifierInputs: IdentifierInput[] = [];
-      const transactions = Object.values(cart)
-        .filter(({ quantity }) => quantity)
-        .map(({ category, quantity, identifierInputs }) => {
-          allIdentifierInputs.push(...identifierInputs);
-          return { category, quantity, identifierInputs };
-        });
-
-      if (transactions.length === 0) {
+      const allCleanedIdentifierInputs: IdentifierInput[] = [];
+      let transactions: any = [];
+      try {
+        transactions = Object.values(cart)
+          .filter(({ quantity }) => quantity)
+          .map(({ category, quantity, identifierInputs }) => {
+            const cleanedIdentifierInputs = cleanIdentifierInputs(
+              identifierInputs
+            );
+            allCleanedIdentifierInputs.push(...cleanedIdentifierInputs);
+            return {
+              category,
+              quantity,
+              identifierInputs: cleanedIdentifierInputs,
+            };
+          });
+        validateIdentifierInputs(allCleanedIdentifierInputs);
+      } catch (error) {
         setCartState("DEFAULT");
-        setCartError(new Error(ERROR_MESSAGE.MISSING_SELECTION));
+        setCartError(error);
         return;
       }
 
-      try {
-        validateIdentifierInputs(allIdentifierInputs);
-      } catch (error) {
+      if (transactions.length === 0) {
```

Comment:
do move this conditional check within the try catch above (line 336), since the `validateIdentifierInputs` is now inside that.

Reply:
done!